azurerm/internal/features/defaults.go

```diff
@@ -19,8 +19,13 @@ func Default() UserFeatures {
 		VirtualMachine: VirtualMachineFeatures{
 			DeleteOSDiskOnDeletion: true,
 			GracefulShutdown:       false,
+			// TODO: this should likely be defaulted true in 3.0
```

Comment:
This should not ever default to true. This is a new feature that should be an opt-in for users.